### PR TITLE
DSD-1809: Prop to display selected item as Menu label

### PR DESCRIPTION
**Pull request by 7emansell**
Fixes JIRA ticket [DSD-1809](https://newyorkpubliclibrary.atlassian.net/browse/DSD-1809).

## This PR does the following:

- Adds `showSelectionAsLabel` as boolean prop
- Updates `aria-label`s to `labelText : selection` instead of just `labelText`, regardless of display

## How has this been tested?

Locally

### Checklist:

- [x] I have updated the Storybook documentation accordingly.
- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### Front End Review:

- [x] Review the Vercel preview deployment once it is ready.

[DSD-1809]: https://newyorkpubliclibrary.atlassian.net/browse/DSD-1809?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/Menu/Menu.test.tsx [hunk 1]
```diff
@@ -109,6 +109,24 @@ describe("Menu allows selection", () => {
...
+        showSelectionAsLabel
+        labelText={"Menu"}
+        listItemsData={defaultListItems}
+      />
+    );
+    const openButton = screen.getByText("Menu");
+    fireEvent.click(openButton);
+    const button1 = screen.getByText("I'm item 1").closest("button");
+    fireEvent.click(button1);
+    fireEvent.mouseDown(document.body);
```
Comment (EdwinGuzman) [inline src/components/Menu/Menu.test.tsx]: Why do you need this?

**Review by EdwinGuzman — APPROVED** [review]
Nice! just one question

File: src/components/Menu/Menu.test.tsx (hunk 1 above)
Comment (7emansell) [reply src/components/Menu/Menu.test.tsx]: Not totally sure which line you mean here, but 125 is just focusing outside of the menu, and lines 126-128 are just confirming that the item is selected and that the label of the item appears twice, first as the menu label and then as the item label. @EdwinGuzman

Comment (EdwinGuzman) [reply src/components/Menu/Menu.test.tsx]: It's for line 125. The menu closes after line 124 and just curious why the need to focus outside of the menu.

File: src/components/Menu/Menu.stories.tsx [hunk 2]
```diff
@@ -405,6 +409,24 @@ export const MenuTypes: Story = {
   },
 };
 
+export const MenuLabel: Story = {
```
Comment (bigfishdesign13) [inline src/components/Menu/Menu.stories.tsx:439]: Can you also add an example where the `selectedItem` prop is set in the code?

File: src/components/Menu/Menu.mdx [hunk 3]
```diff
@@ -59,6 +60,15 @@ Use the icon type when there is limited space such as a table or on mobile.
 
 <Canvas of={MenuStories.MenuTypes} />
 
+## Menu Label
+
+The default label of the menu will be `labelText`. With `showSelectionAsLabel` enabled, the user's selected item will then display as the menu's label. This is useful when the user's choice
```
Comment (bigfishdesign13) [inline src/components/Menu/Menu.mdx]: remove `user's`.

File: src/components/Menu/Menu.mdx [hunk 4]
```diff
@@ -59,6 +60,15 @@ Use the icon type when there is limited space such as a table or on mobile.
...
+should be displayed after selection, like choosing sort/filter criteria.
```
Comment (bigfishdesign13) [inline src/components/Menu/Menu.mdx]: Add...
```
, or if an option needs to be selected when the page first loads.
```

File: src/components/Menu/Menu.test.tsx [hunk 5]
```diff
@@ -109,6 +109,24 @@ describe("Menu allows selection", () => {
     fireEvent.click(openButton);
     expect(button1).toHaveAttribute("data-testid", "selected-item");
   });
+
+  it("displays selected item as label", () => {
```
Comment (bigfishdesign13) [inline src/components/Menu/Menu.test.tsx:112]: Add a test that verifies the label when the `selectedItem` prop is set.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Looks good.  I just have a couple of requests.

**Review by bigfishdesign13 — APPROVED** [review]
Looks good!

**Comment by 7emansell** [comment]
Re-requesting review since conversation with Clare led to new `aria-label`s for the entire component and some more recommendations in the docs

File: .vscode/settings.json [hunk 6]
```diff
@@ -3,5 +3,25 @@
   "editor.formatOnSave": true,
   "editor.defaultFormatter": "esbenp.prettier-vscode",
   "typescript.enablePromptUseWorkspaceTsdk": true,
-  "typescript.tsdk": "node_modules/typescript/lib"
+  "typescript.tsdk": "node_modules/typescript/lib",
```
Comment (EdwinGuzman) [inline .vscode/settings.json]: Hmmm why?

**Review by EdwinGuzman — APPROVED** [review]
Update looks good.

File: .vscode/settings.json (hunk 6 above)
Comment (7emansell) [reply .vscode/settings.json]: Agh this should be in the .gitignore sorry

Comment (EdwinGuzman) [reply .vscode/settings.json]: Well I do like the base settings. I think that's helpful so it updates when you save. I just didn't understand the updates related to this PR. It's fine to remove for now

**Pull request merged**